### PR TITLE
Zlef/en 33066/use high level rest es client

### PR DESCRIPTION
**Pull request by zachlefevre**
- [x] styleCheck passes
- [x] unit tests pass
- [x] integration tests pass
- [x] manual tests pass
~~- [ ] API docs updated~~

### CODE REVIEW

File: spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala [hunk 1]
```diff
@@ -60,14 +167,7 @@ class SpandexElasticSearchClient(
       password: Option[String] = None // scalastyle:on
   ) = {
     this(
-      new PreBuiltXPackTransportClient(
-        Settings.builder().put("cluster.name", clusterName)
-          .put("client.transport.sniff", true)
-          .put("xpack.security.user", "%s:%s".format(userName.getOrElse(" "), password.getOrElse(" ")))
```
Comment (aynleslie) [inline spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala]: this was the only place the username and password were being used.  How is auth being handled now?

File: spandex-integration-tests/src/test/scala/com/socrata/spandex/data/LoaderSpec.scala [hunk 2]
```diff
@@ -18,6 +18,7 @@ class LoaderSpec extends FunSuiteLike
   val datasetId1 = "brown_corpus1"
   val datasetId2 = "brown_corpus2"
 
+  //todo: re-enable. This test takes very long.
```
Comment (aynleslie) [inline spandex-integration-tests/src/test/scala/com/socrata/spandex/data/LoaderSpec.scala]: you look to have re-enabled these. were these todos supposed to be removed?

File: spandex-secondary/docker/Dockerfile [hunk 3]
```diff
@@ -26,7 +26,7 @@ ENV LOG_LEVEL INFO
 ENV JAVA_MAX_METASPACE 64m
 
 ENV SPANDEX_SECONDARY_ES_CLUSTER_NAME spandex
-ENV SPANDEX_SECONDARY_ES_PORT 9300
+ENV SPANDEX_SECONDARY_ES_PORT 9200
```
Comment (aynleslie) [inline spandex-secondary/docker/Dockerfile]: This is well and good, but it's overridden by marathon and I suspect this is why I got this error when trying to deploy your changes:

```
Not a valid protocol version: This is not a HTTP port
```

It's overridden [here](https://github.com/socrata/apps-marathon/blob/master/params/staging/spandex-http.toml#L17), for example.  That `"autocomplete_elasticsearch.host"` bit is taken from [this corresponding file](https://github.com/socrata/apps-marathon/blob/master/params/staging/shared/hosts.toml#L93).

In order to have this change go out cleanly to each env, we'll have to update the shared/hosts.toml file in each env just prior to the deploy.

File: project/Build.scala [hunk 4]
```diff
@@ -190,8 +191,9 @@ object Deps {
     "com.typesafe" %% "scalalogging-slf4j" % "1.1.0",
     "commons-codec" % "commons-codec" % "1.10",
     "commons-io" % "commons-io" % "2.4",
-    "org.elasticsearch" % "elasticsearch" % "5.4.1",
-    "org.elasticsearch.client" % "x-pack-transport" % "5.4.1"
+    "org.elasticsearch" % "elasticsearch" % ElasticsearchVersion,
+    "org.elasticsearch.client" % "x-pack-transport" % ElasticsearchVersion,
```
[the remark this answers is not in the document]
Comment (zachlefevre) [reply project/Build.scala]: No, we definitely do not.

File: spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala [hunk 5]
```diff
@@ -39,15 +39,122 @@ import com.socrata.spandex.common.{ElasticSearchConfig, ElasticsearchClientLogge
...
+
+object UrlHelper{
+  def urlify(url: String, postfix: String): String = urlify(url) + s"/$postfix"
+  def urlify(url: String): String = s"/$url"
+  def entity = (x: String) => new NStringEntity(x, ContentType.APPLICATION_JSON)
+}
+
+
+@JsonKeyStrategy(Strategy.Underscore)
+case class temp_Token(
```
[the remark this answers is not in the document]
Comment (zachlefevre) [reply spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala]: You are right, I no longer need this

File: spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala [hunk 6]
```diff
@@ -414,31 +523,35 @@ class SpandexElasticSearchClient(
...
 
     val timeout = new TimeValue(dataCopyTimeout)
-    val request = client.prepareSearch(indexName)
-      .addSort(FieldSortBuilder.DOC_FIELD_NAME, SortOrder.ASC)
-      .setQuery(queryBuilder)
-      .setTypes(types: _*)
-      .setScroll(timeout)
-      .setSize(calculateScrollSize(dataCopyBatchSize, flushCache))
+
+    val ssb = new SearchSourceBuilder()
```
[the remark this answers is not in the document]
Comment (zachlefevre) [reply spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala]: I gave these the name searchSourceBuilder

File: spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (zachlefevre) [reply spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala]: My mistake. This wasn't necessary.

File: spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala [hunk 7]
```diff
@@ -39,15 +39,122 @@ import com.socrata.spandex.common.{ElasticSearchConfig, ElasticsearchClientLogge
...
+      .performRequest(HttpVerbs.POST, UrlHelper.urlify(indexName, refreshEndpoint))
+
+
+  class HealthResponse(resp: Response){
+    private def getMap(resp: Response) = {
+      val healthStream = resp.getEntity.getContent
+      val healthMap = XContentHelper.convertToMap(XContentType.JSON.xContent(), healthStream, true).asScala
+      healthMap
+    }
+    def getActivePrimaryShards: Int = {
```
[the remark this answers is not in the document]
Comment (zachlefevre) [reply spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala]: I agree

File: spandex-integration-tests/src/test/scala/com/socrata/spandex/data/LoaderSpec.scala (hunk 2 above)
Comment (zachlefevre) [reply spandex-integration-tests/src/test/scala/com/socrata/spandex/data/LoaderSpec.scala]: yes...

File: spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala [hunk 8]
```diff
@@ -39,15 +39,122 @@ import com.socrata.spandex.common.{ElasticSearchConfig, ElasticsearchClientLogge
...
+    val response = lowLevelRestClient.performRequest(HttpVerbs.POST, UrlHelper.urlify(indexName, analyzeEndpoint), emptyJMap, entity)
+    val respStream = response.getEntity.getContent
+    val respString = scala.io.Source.fromInputStream(respStream).mkString
+    val tokens = JsonUtil.parseJson[Tokens](respString).right.get
+    tokens.tokens
+  }
+}
+
+object IndexHelper{
+  def delete(lowLevelRestClient: RestClient)(indexName: String): Unit ={
```
[the remark this answers is not in the document]
Comment (zachlefevre) [reply spandex-common/src/main/scala/com/socrata/spandex/common/client/SpandexElasticSearchClient.scala]: Yes, all of these will be deleted. That being said, createIndex is used within the search client

**Comment by zachlefevre** [comment]
See apps-marathon PR 874

**Pull request merged**